### PR TITLE
Add aws-parallelcluster 2.9.1 and 2.9.0 releases

### PR DESCRIPTION
**Pull request by enrico-usai**
Add missing 2.9.0 version and the conditions for all the past versions of the packages.

### CODE REVIEW

**Comment by adamjstewart** [comment]
I see a lot of overlap with this PR and #18643. Maybe both of you can work together on a single PR? Also, it helps to include the package name in the commit message to prevent these duplicate PRs.

**Comment by enrico-usai** [comment]
@adamjstewart Do you think it is a good approach to remove all the old versions and just keep the most recent one?
In this way we don't have to maintain the list of dependencies for older versions and the package.py stops to grow indefinitely.

BTW patch updated with all the conditions for all the versions.

**Comment by adamjstewart** [comment]
> @adamjstewart Do you think it is a good approach to remove all the old versions and just keep the most recent one?

In the past, we've tended to keep old versions around in case anyone needs to reproduce older installations. I opened #15896 to start a discussion about how to deprecate things in the future, but until that gets resolved it's probably better to keep old versions for now. Of course, you're the maintainer of the package, so you get to make the final decision.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
